core/src/js/components/main-nav/Nav.js

```diff
@@ -265,15 +270,21 @@ export default class Nav {
     if (focusOnFirst) {
       this.focusOn('first'); // Focus on the first top level link
     }
+    // alert others the mobile nav has opened
+    this.elem.dispatchEvent(this.openEvent);
   }
 
   /**
    * Mark this mobile closed, and restore the button text to what it was
    * initially.
    */
   closeMobileNav() {
-    this.setExpanded('false');
-    this.toggle.innerText = this.toggleText;
+    if (this.isExpanded()) {
```

Comment:
Why do we need to check for expanded here? Does it matter?

Reply:
I didn't have it before. But now that we're firing an event when the nav closes, yes, I think it matters:

- Firing an event is much more expensive than DOM tweaking. I don't think we should incur that cost unless we need to.
- If someone is listening for the `closeNav` event, I think they'll only want to be notified if the nav actually closed. Right now we call `closeMobileNav()` anytime anyone clicks outside of a nav. We don't want to fire `closeNav` every time that happens.

Reply:
Sounds like a good idea to me. Out of curiosity, what did you use to measure the expense of the event call?

Reply:
> Out of curiosity, what did you use to measure the expense of the event call?

Intuition. Plus, I watched my event handlers getting called **WAY** more often than they needed to be. :)